students/Kate_Koehn/Session08/class_circle.py

```diff
@@ -0,0 +1,73 @@
+from math import pi
+
+
+class Circle:
+
+    def __init__(self, radius):
+        self.radius = radius
+
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+
+    @property
+    def area(self):
+        return pi * self.radius ** 2
+
+
+    def __str__(self):
+        return f"Circle with radius: {self.radius:.6f}"
+
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.radius})"
+
+
+    def __add__(self, other):
+        return f"{self.__class__.__name__}({self.radius + other.radius})"
+
+
+    def __mul__(self, other):
+        return f"{self.__class__.__name__}({self.radius * other})"
+
+
+    def __rmul__(self, other):
+        return f"{self.__class__.__name__}({other * self.radius})"
+
+
+    def __eq__(self, other):
+        return self.radius == other.radius
+
+
+    def __lt__(self, other):
+        return self.radius < other.radius
+
+
+    def __gt__(self, other):
+        return self.radius > other.radius
+
+
+    def sort(self):
+        return self.radius.sort()
+
+
+class Sphere(Circle):
+
+    def __init__(self, radius):
```

Comment:
non need to write an __init__ if all it's going to do is call super().__init__ -- that will get called be default if you don't define one here.